lib/data/process-test-directory.js

```diff
@@ -597,29 +593,96 @@ ${rows}
     }
   }
 
-  function validateCSVKeys(result) {
-    for (const row of result) {
-      if (typeof row.refId !== 'string' || typeof row.value !== 'string')
-        log.error(
-          `ERROR: References CSV file processing failed: ${referencesCsvFilePath}. Ensure rows are properly formatted.`
-        );
+  // intended to be an internal helper to reduce some code duplication and make logging for csv errors simpler
+  async function readCSVFile(filePath, rowValidator = identity => identity) {
+    const rawCSV = await readCSV(testPlanRecord.find(filePath));
+    let index = 0;
+    function printError(message) {
+      // line number is index+2
+      log.warning(
+        `WARNING: Error parsing ${path.join(testPlanDirectory, filePath)} line ${
+          index + 2
+        }: ${message}`
+      );
+    }
+    try {
+      const firstRowKeysLength = Object.keys(rawCSV[0]).length;
+      for (; index < rawCSV.length; index++) {
+        if (!rowValidator(rawCSV[index])) {
+          printError('validator returned false result');
+          return;
+        }
+        const keysLength = Object.keys(rawCSV[index]).length;
+        if (keysLength != firstRowKeysLength) {
```

Comment:
It might be better to do this before invoking the validator. The CSV parser generates a name for extra columns, e.g. `_10`, and if the validator runs first, it will report an error which the user probably won't understand, e.g.

>     WARNING: Error parsing /home/mike/projects/bocoup/facebook-aria/aria-at/tests/alert/data/commands.csv line 3: Error: Unknown commands.csv key: _10 - check header row?

Reply:
https://github.com/w3c/aria-at/pull/980/commits/7b671cee651e396f9c9cf90420ba5978e8ded4f9